crucible-llvm-syntax/src/Lang/Crucible/LLVM/Syntax/Overrides/String.hs

```diff
@@ -0,0 +1,199 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ImplicitParams #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | Overrides related to C strings
+module Lang.Crucible.LLVM.Syntax.Overrides.String
+  ( readBytesOverride
+  , readCStringOverride
+  , writeBytesOverride
+  , writeCStringOverride
+  ) where
+
+import Control.Monad.IO.Class (liftIO)
+import Data.BitVector.Sized qualified as BV
+import Data.ByteString qualified as BS
+import Data.Parameterized.Context qualified as Ctx
+import Data.Parameterized.NatRepr qualified as DPN
+import Data.Text qualified as DT
+import Data.Text.Encoding qualified as DTE
+import Data.Text.Encoding.Error qualified as DTEE
+import Data.Vector qualified as Vec
+import Lang.Crucible.Backend qualified as LCB
+import Lang.Crucible.LLVM.MemModel qualified as LCLM
+import Lang.Crucible.LLVM.MemModel.Strings qualified as LCLMS
+import Lang.Crucible.Simulator qualified as C
+import Lang.Crucible.Simulator qualified as LCS
+import Lang.Crucible.Types qualified as LCT
+import What4.Interface qualified as WI
+
+-- | Override for @read-bytes@.
+--
+-- The loaded string must be concrete. If a symbolic character is encountered
+-- while loading, this function will generate an assertion failure.
+readBytesOverride ::
```

Comment:
It might be worth adding a TODO here (and near `readBytes`) citing #1061. Although `readBytesOverride` and `readBytes` are currently limited to loading concrete strings, I think this restriction would be lifted if #1061 were resolved.